### PR TITLE
Add confirm plugin

### PR DESCRIPTION
**Pull request by brianpursley**
# Kubectl Confirm Plugin

Kubectl Confirm is a plugin for Kubectl that displays information and asked for confirmation before executing a command.

The following information is displayed:
* Configuration: Context name, Cluster, User, and Namespace
* Dry Run Output (if the executed command supports the `--dry-run` flag)
* Diff Output (if the executed command supports the `--dry-run` and `--output` flags)

## Example Output
```
$ kubectl confirm apply -f ~/changed.yaml
========== Config ===========
Context:    kind-kind
Cluster:    kind-kind
User:       kind-kind
Namespace:  default

========== Dry Run ==========
deployment.apps/foo configured (server dry run)

========== Diff =============
diff -u -N /tmp/LIVE-2275701238/apps.v1.Deployment.default.foo /tmp/MERGED-1055657464/apps.v1.Deployment.default.foo
--- /tmp/LIVE-2275701238/apps.v1.Deployment.default.foo	2022-07-28 10:27:25.690604172 -0400
+++ /tmp/MERGED-1055657464/apps.v1.Deployment.default.foo	2022-07-28 10:27:25.694604038 -0400
@@ -6,7 +6,7 @@
     kubectl.kubernetes.io/last-applied-configuration: |
       {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"name":"foo","namespace":"default"},"spec":{"replicas":1,"revisionHistoryLimit":10,"selector":{"matchLabels":{"app":"foo"}},"template":{"metadata":{"labels":{"app":"foo"}},"spec":{"containers":[{"command":["sh","-c","echo Container bar is running! \u0026\u0026 sleep 9999999"],"image":"busybox:1.30","name":"bar"}]}}}}
   creationTimestamp: "2022-07-28T11:43:58Z"
-  generation: 1
+  generation: 2
   managedFields:
   - apiVersion: apps/v1
     fieldsType: FieldsV1
@@ -90,7 +90,7 @@
 spec:
   progressDeadlineSeconds: 600
   replicas: 1
-  revisionHistoryLimit: 10
+  revisionHistoryLimit: 11
   selector:
     matchLabels:
       app: foo

========== Confirm ==========
The following command will be executed:
kubectl apply -f /home/bpursley/changed.yaml

Enter 'yes' to continue: 
```

If you enter `yes` (exactly) then it will proceed:
```
Enter 'yes' to continue: yes

deployment.apps/foo configured
```

If you enter anything other than `yes` (exactly), then it will stop:
```
Enter 'yes' to continue: no

Command aborted.
```

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Comment by ahmetb** [comment]
I believe we have several plugins similar to this already. Have you checked them out @brianpursley?

It seems like the functionality here is offering net new features, however. 

It might be worthy of capturing diffs in the shortDescription and maybe also explain in description which commands this plugin supports.

**Comment by brianpursley** [comment]
> I believe we have several plugins similar to this already. Have you checked them out @brianpursley?
> 
> It seems like the functionality here is offering net new features, however.
> 
> It might be worthy of capturing diffs in the shortDescription and maybe also explain in description which commands this plugin supports.

Thanks @ahmetb.  You're right, there is some crossover in functionality with the prompt plugin, but there are key differences like dry-run and diff. This plugin will perform a dry-run and diff automatically. I updated shortDescription and description, as you suggested.

See if these changes provide a better description.

File: plugins/confirm.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: confirm
+spec:
+  version: v0.1.0
+  homepage: https://github.com/brianpursley/kubectl-confirm
+  shortDescription: Show config, dry-run, and diff, and prompt for confirmation before running a command
```
Comment (ahmetb) [inline plugins/confirm.yaml:8]: This might not fit "kubectl krew search". Please keep <=50 chars.

Comment (brianpursley) [reply plugins/confirm.yaml]: Ah yeah, I forgot about the limit there, thanks for reminding me.

I changed it to `Dry-run / diff / confirm before running a command`

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!

Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Pull request merged**